make fails => cannot compile C++ source on Mac

I get the following error when I run cmake and then make.  Please advise.  Thanks!

build> cmake ..
CMake Warning (dev) at CMakeLists.txt:1 (project):
  cmake_minimum_required() should be called prior to this top-level project()
  call.  Please see the cmake-commands(7) manual for usage documentation of
  both commands.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at CMakeLists.txt:2 (cmake_minimum_required):
  Compatibility with CMake < 2.4 is not supported by CMake >= 3.0.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Deprecation Warning at CMakeLists.txt:2 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


Compiler with C++14 support found.
-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/fac_rubin/Downloads/ranger-master/cpp_version/build
build> make
**[  6%] Building CXX object CMakeFiles/ranger.dir/src/main.o
/Users/fac_rubin/Downloads/ranger-master/cpp_version/src/main.cpp:59:96: error: too few arguments to function call, expected 31, have 30
      arg_handler.randomsplits, arg_handler.maxdepth, arg_handler.regcoef, arg_handler.usedepth);
                                                                                               ^
/Users/fac_rubin/Downloads/ranger-master/cpp_version/src/Forest/Forest.h:41:8: note: 'initCpp' declared here
  void initCpp(std::string dependent_variable_name, MemoryMode memory_mode, std::string input_file, uint mtry,
       ^
1 error generated.
make[2]: *** [CMakeFiles/ranger.dir/src/main.o] Error 1
make[1]: *** [CMakeFiles/ranger.dir/all] Error 2
make: *** [all] Error 2**
build>

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -1,8 +1,8 @@
 Package: ranger
 Type: Package
 Title: A Fast Implementation of Random Forests
-Version: 0.15.2
-Date: 2023-07-02
+Version: 0.15.3
+Date: 2023-07-19
 Author: Marvin N. Wright [aut, cre], Stefan Wager [ctb], Philipp Probst [ctb]
 Maintainer: Marvin N. Wright <cran@wrig.de>
 Description: A fast implementation of Random Forests, particularly suited for high

--- a/NEWS.md
+++ b/NEWS.md
@@ -1,4 +1,7 @@
 
+# ranger 0.15.3
+* Fix min bucket option in C++ version
+
 # ranger 0.15.2
 * Fix memory error for always.split.variables in certain settings
 

--- a/cpp_version/src/main.cpp
+++ b/cpp_version/src/main.cpp
@@ -52,7 +52,7 @@ void run_ranger(const ArgumentHandler& arg_handler, std::ostream& verbose_out) {
   // Call Ranger
   forest->initCpp(arg_handler.depvarname, arg_handler.memmode, arg_handler.file, arg_handler.mtry,
       arg_handler.outprefix, arg_handler.ntree, &verbose_out, arg_handler.seed, arg_handler.nthreads,
-      arg_handler.predict, arg_handler.impmeasure, arg_handler.targetpartitionsize, arg_handler.splitweights,
+      arg_handler.predict, arg_handler.impmeasure, arg_handler.targetpartitionsize, arg_handler.minbucket, arg_handler.splitweights,
       arg_handler.alwayssplitvars, arg_handler.statusvarname, arg_handler.replace, arg_handler.catvars,
       arg_handler.savemem, arg_handler.splitrule, arg_handler.caseweights, arg_handler.predall, arg_handler.fraction,
       arg_handler.alpha, arg_handler.minprop, arg_handler.holdout, arg_handler.predictiontype,

--- a/cpp_version/src/utility/ArgumentHandler.cpp
+++ b/cpp_version/src/utility/ArgumentHandler.cpp
@@ -23,7 +23,7 @@ ArgumentHandler::ArgumentHandler(int argc, char **argv) :
     caseweights(""), depvarname(""), fraction(0), holdout(false), memmode(MEM_DOUBLE), savemem(false), skipoob(false), predict(
         ""), predictiontype(DEFAULT_PREDICTIONTYPE), randomsplits(DEFAULT_NUM_RANDOM_SPLITS), splitweights(""), nthreads(
         DEFAULT_NUM_THREADS), predall(false), alpha(DEFAULT_ALPHA), minprop(DEFAULT_MINPROP), maxdepth(
-        DEFAULT_MAXDEPTH), file(""), impmeasure(DEFAULT_IMPORTANCE_MODE), targetpartitionsize(0), mtry(0), outprefix(
+        DEFAULT_MAXDEPTH), file(""), impmeasure(DEFAULT_IMPORTANCE_MODE), targetpartitionsize(0), minbucket(0), mtry(0), outprefix(
         "ranger_out"), probability(false), splitrule(DEFAULT_SPLITRULE), statusvarname(""), ntree(DEFAULT_NUM_TREE), replace(
         true), verbose(false), write(false), treetype(TREE_CLASSIFICATION), seed(0), usedepth(false) {
   this->argc = argc;
@@ -33,7 +33,7 @@ ArgumentHandler::ArgumentHandler(int argc, char **argv) :
 int ArgumentHandler::processArguments() {
 
   // short options
-  char const *short_options = "A:C:D:F:HM:NOP:Q:R:S:U:XZa:b:c:d:f:hi:j:kl:m:o:pr:s:t:uvwy:z:";
+  char const *short_options = "A:C:D:F:HM:NOP:Q:R:S:U:XZa:b:c:d:f:hi:j:kl:m:n:o:pr:s:t:uvwy:z:";
 
 // long options: longname, no/optional/required argument?, flag(not used!), shortname
     const struct option long_options[] = {
@@ -53,7 +53,6 @@ int ArgumentHandler::processArguments() {
       { "nthreads",             required_argument,  0, 'U'},
       { "predall",              no_argument,        0, 'X'},
       { "version",              no_argument,        0, 'Z'},
-
       { "alpha",                required_argument,  0, 'a'},
       { "minprop",              required_argument,  0, 'b'},
       { "catvars",              required_argument,  0, 'c'},
@@ -65,6 +64,7 @@ int ArgumentHandler::processArguments() {
       { "usedepth",             no_argument,        0, 'k'},
       { "targetpartitionsize",  required_argument,  0, 'l'},
       { "mtry",                 required_argument,  0, 'm'},
+      { "minbucket",            required_argument,  0, 'n'},
       { "outprefix",            required_argument,  0, 'o'},
       { "probability",          no_argument,        0, 'p'},
       { "splitrule",            required_argument,  0, 'r'},
@@ -306,6 +306,20 @@ int ArgumentHandler::processArguments() {
       }
       break;
 
+    case 'n':
+      try {
+        int temp = std::stoi(optarg);
+        if (temp < 1) {
+          throw std::runtime_error("");
+        } else {
+          minbucket = temp;
+        }
+      } catch (...) {
+        throw std::runtime_error(
+            "Illegal argument for option 'minbucket'. Please give a positive integer. See '--help' for details.");
+      }
+      break;
+      
     case 'o':
       outprefix = optarg;
       break;
@@ -573,6 +587,7 @@ void ArgumentHandler::displayHelp() {
   std::cout << "    " << "                              (Default: sqrt(p) with p = number of independent variables)"
       << std::endl;
   std::cout << "    " << "--targetpartitionsize N       Set minimal node size to N." << std::endl;
+  std::cout << "    " << "--minbucket N                 Set min bucket size to N." << std::endl;
   std::cout << "    "
       << "                              For Classification and Regression growing is stopped if a node reaches a size smaller than N."
       << std::endl;

--- a/cpp_version/src/utility/ArgumentHandler.h
+++ b/cpp_version/src/utility/ArgumentHandler.h
@@ -71,6 +71,7 @@ public:
   std::string file;
   ImportanceMode impmeasure;
   uint targetpartitionsize;
+  uint minbucket;
   uint mtry;
   std::string outprefix;
   bool probability;

--- a/cpp_version/src/version.h
+++ b/cpp_version/src/version.h
@@ -1,3 +1,3 @@
 #ifndef RANGER_VERSION
-#define RANGER_VERSION "0.15.2"
+#define RANGER_VERSION "0.15.3"
 #endif

